Create datatype_tabular.md

Beachyesh opened the pull request:
Proposing this file be added to galaxy FAQs. Please suggest any changes that need to be made, including adding any contributor names.

Beachyesh commented:
@hexylena How can I fix this failed check

hexylena commented:
hi @Beachyesh for fixing this error (I guess you deleted the comment), please add yourself to the CONTRIBUTORS.yaml file.

hexylena commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,31 @@
+contributors: [beachyesh] 
+---
+
+[Back to Support Hub](https://galaxyproject.org/support/)
```
please do not include this.

hexylena commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,31 @@
+
+[Back to Support Hub](https://galaxyproject.org/support/)
+
+Related Topics 
+- [Fastq Manipulation and Quality Control](https://galaxyproject.org/tutorials/ngs/#fastq-manipulation-and-quality-control)  
+- [How to format fastq data for tools that require .fastqsanger format?](https://galaxyproject.org/support/fastqsanger/)
+- [Understanding compressed fastq data (fastq.gz)](https://galaxyproject.org/support/compressed-fastq/)
+- [Common datatypes explained](https://galaxyproject.org/learn/datatypes/)
+- [Input datatype misassignment and errors](https://galaxyproject.org/support/tool-error/)
```
this is interesting but I'm not sure we should include this especially when e.g. `Understanding compressed fastq data (fastq.gz)` will probably become a faq sometday.

I would suggest removing this

shiltemann commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,22 @@
+---
+title: Datatype Tabular
+area: Getting inputs right    # FAQs will be grouped by these areas on the FAQ page
```
please try to align with existing areas, this sounds more like a title

shiltemann commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,22 @@
+area: Getting inputs right    # FAQs will be grouped by these areas on the FAQ page
+description: Format help for Tabular/BED/Interval Datasets
+layout: faq          
+
```
please remove this empty line, it may break the rendering

shiltemann commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,22 @@
+
+---
+ 
+### Help for Tabular Datasets
```
you don't need a title here, the title defined in the metadata above will be placed at the top of the FAQ

Beachyesh commented:
@shiltemann Got it, I'll make all the necessary changes. Thanks!

Beachyesh commented:
@hexylena @shiltemann I updated the file to make it shorter and more aligned with how the rest of the faqs have been written. Let me know about further changes. Thank you for your input! :)

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,21 @@
+---
+title: Datatype Tabular
```
Could we have a title here that indicates what this FAQ is trying to address?

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,21 @@
+---
+ 
+
+- A Tabluar datatype is human readable and has tabs seperating data columns.
```
```suggestion
- A Tabular datatype is human readable and has tabs separating data columns.
```

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,21 @@
+  - Note: tabular data is different from comma seperated data (.csv)
+- Common tabular datatypes are .bed, .gtf, .interval, or .txt.
+- The datatype metadata attribute can be directly reassigned to tabular format data.
+- Click the {% icon galaxy-pencil %} icon to reach the **_Edit Attributes_** form. 
```
I think that this is where the FAQ starts and we could have the numbered steps

shiltemann replied:
It is best if we also write out the icon description in text, followed by the icon itself, for increased accessibility (e.g. for screen readers)

```suggestion
- Click the pencil icon {% icon galaxy-pencil %} to reach the **_Edit Attributes_** form. 
```

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,21 @@
+- If the required input is a BED or Interval datatype, adjusting (.tab → .bed, .tab → .interval) maybe possible using a combination of **_Text Manipulation_** tools, to create a dataset that matches the BED or Interval datatype specifications.
+
+#### Tips
+- Some tools require that BED format is followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.
+- These tools will fail if run with malformed BED datasets or non-specific column assignments.
```
do you think that these tips could be included within the previous bullet point?

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,21 @@
+- Some tools require that BED format is followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.
+- These tools will fail if run with malformed BED datasets or non-specific column assignments.
+
+**Solution**: Reorganize the data to be in BED format and rerun. 
```
is this also a step in the FAQ?

shiltemann commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,21 @@
+- Common tabular datatypes are .bed, .gtf, .interval, or .txt.
+- The datatype metadata attribute can be directly reassigned to tabular format data.
+- Click the {% icon galaxy-pencil %} icon to reach the **_Edit Attributes_** form. 
+  - Change the datatype (3rd tab) and save, then label columns (1st tab) and save.  Metadata will assign, then the dataset can be used.
```
```suggestion
  - Change the datatype (3rd tab) and save, then label columns (1st tab) and save.  Metadata will be assigned, then the dataset can be used.
```

shiltemann reviewed (commented):
Thanks @Beachyesh!

Beachyesh commented:
@beatrizserrano I have updated the file.  Added a title, numbered the steps, and removed the tip to add a new bullet point. 
@shiltemann I have made the suggested commits as well. 
Always open to further improvement. Thanks to both of you for helping me out ❤️

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+  - Note: tabular data is different from comma seperated data (.csv)
+1. Common tabular datatypes are `.bed`, `.gtf`, `.interval`, or `.txt`.
+2. The datatype metadata attribute can be directly reassigned to tabular format data.
+3. Click the {% icon galaxy-pencil %} ✏️ icon to reach the **_Edit Attributes_** form. 
```
This `{% icon galaxy-pencil %}` translates into ✏️, so there's no need to add ✏️

shiltemann replied:
We try to always also include the text description of the icon, followed by the icon itself, for accessibility reasons

```suggestion
3. Click the pencil icon {% icon galaxy-pencil %} to reach the **_Edit Attributes_** form. 
```

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+2. The datatype metadata attribute can be directly reassigned to tabular format data.
+3. Click the {% icon galaxy-pencil %} ✏️ icon to reach the **_Edit Attributes_** form. 
+   - Change the datatype (3rd tab) and save
+   - label columns (1st tab) and save
```
```suggestion
   - Label columns (1st tab) and save
```

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+3. Click the {% icon galaxy-pencil %} ✏️ icon to reach the **_Edit Attributes_** form. 
+   - Change the datatype (3rd tab) and save
+   - label columns (1st tab) and save
+   -  Metadata will be assigned, then the dataset can be used.
```
```suggestion
   - Metadata will be assigned, then the dataset can be used.
```

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+- A Tabular datatype is human readable and has tabs separating data columns.
+  - Note: tabular data is different from comma seperated data (.csv)
+1. Common tabular datatypes are `.bed`, `.gtf`, `.interval`, or `.txt`.
+2. The datatype metadata attribute can be directly reassigned to tabular format data.
```
I have a hard time understanding this sentence, could you rephrase it?

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+---
+title: Indentifying and Formatting Tabular Dataset 
```
```suggestion
title: Identifying and formatting Tabular Datasets
```

beatrizserrano replied:
@Beachyesh this commit was not done completely, do you know that you can do it directly from this comment here by clicking on "Commit suggestion" instead of editing the file?

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+
+- A Tabular datatype is human readable and has tabs separating data columns.
+  - Note: tabular data is different from comma seperated data (.csv)
+1. Common tabular datatypes are `.bed`, `.gtf`, `.interval`, or `.txt`.
```
I think that this is a clarification, but not a proper step, same with step 2

Beachyesh replied:
Noted.

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,20 @@
+1. Common tabular datatypes are `.bed`, `.gtf`, `.interval`, or `.txt`.
+2. The datatype metadata attribute can be directly reassigned to tabular format data.
+3. Click the pencil icon {% icon galaxy-pencil %} to reach the **_Edit Attributes_** form. 
+   - Change the datatype (3rd tab) and save
+   - Label columns (1st tab) and save
+   - Metadata will be assigned, then the dataset can be used.
```
these 3 could maybe be converted into steps

Beachyesh replied:
Done.

Beachyesh commented:
@shiltemann @hexylena I have made the suggested commits to these FAQs. Please let me know if there are any further suggestions for this file. Thanks!

Beachyesh commented:
@beatrizserrano I apologize for my silly mistake, I just kept reading "Indentifying" as "Identifying" and there was some issue with making the suggested commit the previous time. Just a mistake on my part 🙈

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,17 @@
+   3. Metadata will be assigned, then the dataset can be used.
+2. If the required input is a BED or Interval datatype, adjusting (``.tab`` → ``.bed``, ``.tab`` → ``.interval``) maybe possible using a combination of **_Text Manipulation_** tools, to create a dataset that matches required specifications.
+3. Some tools require that BED format be followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.
+   - These tools will fail, if they are run with malformed BED datasets or non-specific column assignments.
+   - **Solution** is to reorganize the data to be in BED format and rerun. 
```
```suggestion
   - These tools will fail, if they are run with malformed BED datasets or non-specific column assignments.
   - **Solution**: reorganize the data to be in BED format and rerun. 
```

beatrizserrano commented:
> @beatrizserrano I apologize for my silly mistake, I just kept reading "Indentifying" as "Identifying" and there was some issue with making the suggested commit the previous time. Just a mistake on my part see_no_evil

no worries, I've made a couple of minor suggestions and I think that we can merge very very soon, thanks @Beachyesh!

Beachyesh commented:
Great! Thanks a ton! @beatrizserrano 😄

beatrizserrano commented on faqs/galaxy/datatype_tabular.md:
```diff
@@ -0,0 +1,18 @@
+---
+title: Identifying and formatting Tabular Datasets 
+area: Datatypes
+description: Format help for Tabular/BED/Interval Datasets
+contributors: [jennaj, beachyesh]
+layout: faq          
```
I'm not sure the order of the rows here matters, but just for consistency, could you move the contributors to the end? I think that's how it appears in the other FAQs. Thanks!

Beachyesh replied:
Done. Thanks!

beatrizserrano reviewed (approved):
this looks good, thanks @Beachyesh!
